mllib-local/src/main/scala/org/apache/spark/ml/linalg/Matrices.scala

```diff
@@ -154,11 +172,12 @@ sealed trait Matrix extends Serializable {
  * @param isTransposed whether the matrix is transposed. If true, `values` stores the matrix in
  *                     row major.
  */
-class DenseMatrix (
-    val numRows: Int,
-    val numCols: Int,
-    val values: Array[Double],
-    override val isTransposed: Boolean) extends Matrix {
+@Since("2.0.0")
+class DenseMatrix @Since("2.0.0") (
```

Comment:
This might be overkill, since the class is marked "since 2.0.0". Everything in it is implicitly "since 2.0.0" as well -- or later. But right now there's no later version.